source/common/upstream/cluster_manager_impl.cc

```diff
@@ -502,10 +502,16 @@ void ClusterManagerImpl::loadCluster(const envoy::api::v2::Cluster& cluster, boo
   // If an LB is thread aware, create it here. The LB is not initialized until cluster pre-init
   // finishes.
   if (cluster_reference.info()->lbType() == LoadBalancerType::RingHash) {
+    if (cluster.has_lb_subset_config()) {
```

Comment:
I think it is supported? I definitely added code to Subset LB to make them work? What broke exactly? cc @zuercher

Reply:
I don't see how it can work. We have https://github.com/envoyproxy/envoy/blob/a87d265d1c1819926f2b2e8e7e866244fd40f574/source/common/upstream/cluster_manager_impl.cc#L764 (which the proto fuzzer found), the thread aware LB has this factory created.

Reply:
https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/subset_lb.cc#L400

? That code is all covered somehow since I had to add it. So I'm confused. Can you take a look at the discrepancy?

Reply:
The code is covered, but my best guess is that this is via https://github.com/envoyproxy/envoy/blob/master/test/common/upstream/subset_lb_test.cc, which doesn't exercise the full config ingrestion path via cluster manager, but is instead focused on `subset_lb.cc`. 

Do we have any known success of this working end-to-end since the thread-aware patch?

Reply:
I don't know. But I don't see why it wouldn't work, since we setup the the thread local context on each thread for subset LB.

I guess my main point is I don't mind blocking this, but we should delete the code and tests from subset LB if we aren't supporting it.

Reply:
Actually, I think this does work, validated by hand, will just remove the `ASSERT`.